ChartsRealm/Classes/Data/RealmBarDataSet.swift

```diff
@@ -197,9 +197,18 @@ open class RealmBarDataSet: RealmBarLineScatterCandleBubbleDataSet, IBarChartDat
     /// is calculated from the Entries that are added to the DataSet
     fileprivate var _stackSize = 1
     
-    internal override func buildEntryFromResultObject(_ object: RLMObject, x: Double) -> ChartDataEntry
+    internal override func buildEntryFromResultObject(_ object: RLMObjectBase, x: Double) -> ChartDataEntry
     {
-        let value = object[_yValueField!]
+        var value: Any?
+        if let object = object as? RLMObject
```

Comment:
@jjatie Do you have better solution that we could reduce the duplicated code like this pattern.

Basically, we take `RLMObjectBase`, but later need to cast it to a concrete class type to get the value. Objc is using RLMObject, while Swift is using Object.

I don't come up a better one here.

Reply:
first, you should use `let value: Object`. It doesn't need to be optional or a variable because you know you're going to initialize it right away.

Second, you can write it like this
```
let value = (object as? RLMObject)[_yValueField!] ?? (object as! Object)[_yValueField!]
```
or
```
let object = (object as? RLMObject) ?? (object as! Object)
let value = object[_yValueField!]
```

Reply:
seems neither of them works.
first statement gives me `Type 'RLMObject?' has no subscript members`, as the left operand is optional, while the right operand would work.

second statement gives me `Type 'RLMObjectBase' has no subscript members`

Reply:
Oops, should have been
`let value = (object as? RLMObject)?[_yValueField!] ?? (object as! Object)[_yValueField!]`